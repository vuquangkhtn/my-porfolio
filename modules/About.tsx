import Image from 'next/image';
import { useRouter } from 'next/router';
import { Button } from 'common';
import softwareEngineerImg from 'images/undraw/undraw_software_engineer_lvl5.svg';

const About = () => {
  const router = useRouter();
  return <section className="about full-screen d-lg-flex justify-content-center align-items-center" id="about">
    <div className="container">
      <div className="row">

        <div className="col-lg-7 col-md-12 col-12 d-flex align-items-center">
          <div className="about-text">
            <small className="small-text">Welcome to <span className="mobile-block">my portfolio website!</span></small>
            <h1 className="animated animated-text">
              <span className="mr-2">Hey folks, I am</span>
              <div className="animated-info">
                <span className="animated-item">Marvel Sann</span>
                <span className="animated-item">Web Designer</span>
                <span className="animated-item">UI Specialist</span>
              </div>
            </h1>

            <p>Building a successful product is a challenge. I am highly energetic in user experience design, interfaces and web development.</p>

            <div className="custom-btn-group mt-4">
              <Button className="mr-lg-2"><i className='uil uil-file-alt'></i> Download Resume</Button>
              <Button mode="secondary" onClick={() => router.push('#contact')}>Get a free quote</Button>
            </div>
          </div>
        </div>

        <div className="col-lg-5 col-md-12 col-12">
          <div className="about-image svg">
            <Image src={softwareEngineerImg} className="img-fluid" alt="svg image" />
          </div>
        </div>

      </div>
    </div>
  </section >;
};

export default About;
