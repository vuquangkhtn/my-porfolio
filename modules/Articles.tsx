import Image from 'next/image';
import articles from 'api/article';
import imgSrc from 'static/images/placeholder-image.png';

const ArticleList = () => {
  return (
    <div className="row">
      {articles.map((article) => {
        return (
          <div
            className="col-md-6 col-sm-12 col-lg-4 py-3 pe-auto"
            key={article.id}
            onClick={() => window.open(article.url, '_blank')}
          >
            <div className="card mr-1">
              <div className="card-header" style={{ height: '200px', overflow: 'hidden' }}>
                <Image
                  objectFit='contain'
                  src={article.thumbnail || imgSrc}
                  alt={`image-${article.id}`}
                />
              </div>
              <div className="card-body">
                <h3 className="">{article.title}</h3>
                <p className="card-text">{article.description}</p>
              </div>
            </div>
          </div>
        );
      })}
    </div>
  );
};

const Article = () => {
  return (
    <section
      className="resume py-5 d-lg-flex justify-content-center align-items-center"
    >
      <div className="container">
        <div className="row">
          <div className="col-12">
            <h2 className="mb-4">Articles</h2>
            <ArticleList />
          </div>
        </div>
      </div>
    </section>
  );
};

export default Article;
