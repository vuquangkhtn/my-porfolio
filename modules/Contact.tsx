import React, { useState } from 'react';
import { Button, Input, Icon, TextArea, confirmPopup } from 'common';
import { addContact } from 'api/contact';

const Contact = ({ user }: { user: User }) => {
  const [email, setEmail] = useState('');
  const [name, setName] = useState('');
  const [description, setDescription] = useState('');
  const formatedPhone = user.phone?.replace(/[^0-9]+/g, '');
  const encodedAddress = encodeURIComponent(user.address);

  const sendContact = async () => {
    try {
      if (!email || !name) {
        confirmPopup({
          header: 'Send Contact Failed',
          confirmation: 'Please input email and name first!',
          isMessage: true,
        });
        return;
      }

      const isAccepted = await confirmPopup({
        header: 'Send Contact Confirmation',
        confirmation: 'Are you sure to contact me with those information?',
      });
      if (!isAccepted) return;

      const result = await addContact({ email, name, description });
      if (result) {
        confirmPopup({
          header: 'Send Contact Successfully',
          confirmation:
            'Thanks for getting in touch! I sent you an email for confirmation.',
          isMessage: true,
        });
        setEmail('');
        setName('');
        setDescription('');
      } else {
        throw {};
      }
    } catch (e) {
      confirmPopup({
        header: 'Send Contact Failed',
        confirmation:
          'This is my problem. Kindly use other ways to contact me.',
        isMessage: true,
      });
    }
  };

  return (
    <section className="contact py-5" id="contact">
      <div className="container">
        <div className="row">
          <div className="col-lg-5 mr-lg-5 col-12">
            <div className="google-map w-100">
              <iframe
                src={`https://maps.google.com/maps?q=${encodedAddress}&t=&z=13&ie=UTF8&iwloc=&output=embed`}
                width="400"
                height="300"
                frameBorder="0"
                style={{ border: 0 }}
                allowFullScreen
              ></iframe>
            </div>

            <div className="contact-info d-flex justify-content-between align-items-center py-4 px-lg-5">
              <div className="contact-info-item">
                <h3 className="mb-3 text-white">My information</h3>
                <p className="footer-text mb-0">{user.phone}</p>
                <p>
                  <a href={`mailto:${user.email}`}>{user.email}</a>
                </p>
              </div>

              <ul className="social-links">
                {user.github && (
                  <li>
                    <a
                      target="_blank"
                      href={user.github}
                      title="github"
                      rel="noreferrer"
                    >
                      <Icon icon="uil:github" width="24" />
                    </a>
                  </li>
                )}
                {user.linkedin && (
                  <li>
                    <a
                      target="_blank"
                      href={user.linkedin}
                      title="linkedin"
                      rel="noreferrer"
                    >
                      <Icon icon="uil:linkedin" width="24" />
                    </a>
                  </li>
                )}
                {formatedPhone && (
                  <li>
                    <a href={`tel:${formatedPhone}`} title="phone">
                      <Icon icon="uil:phone" width="24" />
                    </a>
                  </li>
                )}
              </ul>
            </div>
          </div>

          <div className="col-lg-6 col-12">
            <div className="contact-form">
              <h2 className="mb-4">
                {'Interested to work together? Let\'s talk'}
              </h2>

              <div>
                <div className="row">
                  <div className="col-lg-6 col-12">
                    <Input
                      autoComplete="nope"
                      type="text"
                      name="name"
                      placeholder="Your Name"
                      onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                        setName(e.target.value)
                      }
                    />
                  </div>

                  <div className="col-lg-6 col-12">
                    <Input
                      autoComplete="nope"
                      type="email"
                      name="email"
                      placeholder="Email"
                      onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                        setEmail(e.target.value)
                      }
                    />
                  </div>

                  <div className="col-12">
                    <TextArea
                      autoComplete="nope"
                      name="message"
                      rows={6}
                      placeholder="Message"
                      onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
                        setDescription(e.target.value)
                      }
                    />
                  </div>

                  <div className="col-lg-5 col-12">
                    <Button mode="submit" onClick={sendContact}>
                      Send Contact
                    </Button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default Contact;
