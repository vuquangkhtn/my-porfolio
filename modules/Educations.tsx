const Educations = ({ educations }) => {
    return (
        <section className="resume py-5 d-lg-flex justify-content-center align-items-center" id="education">
            <div className="container">
                <div className="row">

                    <div className="col-12">
                        <h2 className="mb-4">Educations</h2>
                        <div className="timeline">
                            {educations.map(education => (
                                <div key={education.id} className="timeline-wrapper">
                                    <div className="timeline-yr">
                                        <span>{education.organization}</span>
                                    </div>
                                    <div className="timeline-info">
                                        <h3><span />{education.name}</h3>
                                        <p>{education.description}</p>
                                    </div>
                                </div>
                            ))}
                        </div>
                    </div>

                </div>
            </div>
        </section>
    );
};

export default Educations;
