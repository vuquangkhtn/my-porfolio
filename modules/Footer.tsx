import { Link } from 'my-porfolio-common-library';

const Footer = () => {
  return (
    <footer className="footer py-5">
      <div className="container">
        <div className="row">
          <div className="col-lg-12 col-12">
            <p className="copyright-text text-center">
              Designed by{' '}
              <Link rel="nofollow" href="htwtps://www.facebook.com/tooplate">
                Tooplate
              </Link>
            </p>
          </div>
        </div>
      </div>
    </footer>
  );
};

export default Footer;
