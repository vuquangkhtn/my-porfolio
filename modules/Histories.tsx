import { generateMonths } from 'utils/date';

type HistoryListProps = {
  histories: Array<History>
};
const HistoryList = ({ histories }: HistoryListProps) => {
  return (
    <div className="timeline">
      {histories.map(history => (
        <div key={history.id} className="timeline-wrapper">
          <div className="timeline-yr">
            <span>{generateMonths(history.time, history.nextTime?.time)}</span>
          </div>
          <div className="timeline-info">
            <h3><span />{history.name}<small>{history.company}</small></h3>
            <p>{`From ${history.time} to ${history.nextTime?.time || 'Present'}`}</p>
          </div>
        </div>
      ))}
    </div>
  );
};


type HistoryProps = {
  histories: Array<History>
};
const Histories = ({ histories }: HistoryProps) => {
  if (!histories || histories.length === 0) return null;

  return (
    <section className="resume py-5 d-lg-flex justify-content-center align-items-center" id="history">
      <div className="container">
        <div className="row">

          <div className="col-12">
            <h2 className="mb-4">Histories</h2>
            <HistoryList histories={histories} />
          </div>

        </div>
      </div>
    </section>
  );
};

export default Histories;
