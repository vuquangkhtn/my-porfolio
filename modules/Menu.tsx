import { useEffect, useContext, useState, WheelEvent } from 'react';
import { Icon, NavBarToggler } from 'common';
import ThemeContext from 'context/ThemeContext';
import { checkScrollDirectionIsUp, isScrollBottom, isScrollTop } from 'utils/scroll';

const Menu = ({ username }: { username: string }) => {
  const [isDarkMode, setDarkMode] = useContext(ThemeContext);
  const [menuShown, showMenu] = useState(false);
  const toggleColorMode = () => {
    setDarkMode(!isDarkMode);
  };

  const toggleNavbar = () => {
    showMenu(!menuShown);
  };

  useEffect(() => {
    const checkScrollDirection = (e: any | WheelEvent) => {
      const navBarDOM = document.querySelector('.navbar');
      if (checkScrollDirectionIsUp(e)) {
        navBarDOM?.classList.add('headroom--pinned');
        navBarDOM?.classList.remove('headroom--unpinned');
      } else {
        navBarDOM?.classList.add('headroom--unpinned');
        navBarDOM?.classList.remove('headroom--pinned');
      }
    };

    const checkScrollPosition = () => {
      const navBarDOM = document.querySelector('.navbar');
      if (isScrollTop(document)) {
        navBarDOM?.classList.add('headroom--top');
        navBarDOM?.classList.remove('headroom--not-top');
      } else {
        navBarDOM?.classList.remove('headroom--top');
        navBarDOM?.classList.add('headroom--not-top');
      }

      if (isScrollBottom(document)) {
        navBarDOM?.classList.add('headroom--bottom');
        navBarDOM?.classList.remove('headroom--not-bottom');
      } else {
        navBarDOM?.classList.remove('headroom--bottom');
        navBarDOM?.classList.add('headroom--not-bottom');
      }
    };

    document.addEventListener('wheel', checkScrollDirection, { passive: true });
    document.addEventListener('scroll', checkScrollPosition, { passive: true });
    return () => {
      document.removeEventListener('wheel', checkScrollDirection);
      document.removeEventListener('scroll', checkScrollPosition);
    };
  }, []);

  return (
    <nav className="navbar navbar-expand-sm navbar-light headroom">
      <div className="container">
        <a className="navbar-brand" href=""><Icon icon='uil:user' width="40" /> {username}</a>
        <NavBarToggler className="navbar-toggler" onClick={toggleNavbar} expanded={menuShown} title="Navbar Toggler" />

        <div className={`navbar-collapse ${menuShown ? 'show' : 'collapse'}`}>
          <ul className="navbar-nav mx-auto">
            <li className="nav-item">
              <a href="#skill" className="nav-link"><span data-hover="Skill">Skill</span></a>
            </li>
            <li className="nav-item">
              <a href="#history" className="nav-link"><span data-hover="History">History</span></a>
            </li>
            <li className="nav-item">
              <a href="#project" className="nav-link"><span data-hover="Project">Project</span></a>
            </li>
            <li className="nav-item">
              <a href="#contact" className="nav-link"><span data-hover="Contact">Contact</span></a>
            </li>
          </ul>

          <ul className="navbar-nav ml-lg-auto">
            <li className="ml-lg-4">
              <div
                className="color-mode d-lg-flex justify-content-center align-items-center"
                onClick={toggleColorMode}
                data-testid={isDarkMode ? 'dark-icon' : 'light-icon'}
              >
                {isDarkMode ?
                  <Icon icon="uil:sun" width="30" color="white" />
                  :
                  <Icon icon="uil:moon" width="30" color="black" />
                }
                Color mode
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  );
};

export default Menu;
