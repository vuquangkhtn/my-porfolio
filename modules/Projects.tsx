import Image from 'next/image';
import project1Img from 'images/project/project-image01.png';
import project2Img from 'images/project/project-image02.png';
import project3Img from 'images/project/project-image03.png';
import project4Img from 'images/project/project-image04.png';
import project5Img from 'images/project/project-image05.png';

const Projects = () => {
  return (
    <section className="project py-5" id="project">
      <div className="container">

        <div className="row">
          <div className="col-lg-11 text-center mx-auto col-12">

            <div className="col-lg-8 mx-auto">
              <h2>Things I have designed for digital media agencies</h2>
            </div>

            <div className="owl-carousel owl-theme">
              <div className="item">
                <div className="project-info">
                  <Image src={project1Img} className="img-fluid" alt="project image" />
                </div>
              </div>

              <div className="item">
                <div className="project-info">
                  <Image src={project2Img} className="img-fluid" alt="project image" />
                </div>
              </div>

              <div className="item">
                <div className="project-info">
                  <Image src={project3Img} className="img-fluid" alt="project image" />
                </div>
              </div>

              <div className="item">
                <div className="project-info">
                  <Image src={project4Img} className="img-fluid" alt="project image" />
                </div>
              </div>

              <div className="item">
                <div className="project-info">
                  <Image src={project5Img} className="img-fluid" alt="project image" />
                </div>
              </div>
            </div>

          </div>
        </div>
      </div>
    </section>
  );
};

export default Projects;
