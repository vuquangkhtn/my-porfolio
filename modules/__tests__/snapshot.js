import { render } from '@testing-library/react';
import About from '../About';
import Contact from '../Contact';
import Educations from '../Educations';
import Footer from '../Footer';
import Histories from '../Histories';
import Menu from '../Menu';
import Projects from '../Projects';
import Skills from '../Skills';
import Articles from '../Articles';

import mockUser from '../../__mocks__/data/user';
import mockEducations from '../../__mocks__/data/educations';
import mockHistories from '../../__mocks__/data/histories';
import mockExperiences from '../../__mocks__/data/experiences';
import mockSkills from '../../__mocks__/data/skills';

it('should render About', () => {
  const { container } = render(<About user={mockUser} />);
  expect(container).toMatchSnapshot();
});

it('should render Contact', () => {
  const { container } = render(<Contact user={mockUser} />);
  expect(container).toMatchSnapshot();
});

it('should render Educations', () => {
  const { container } = render(<Educations educations={mockEducations} />);
  expect(container).toMatchSnapshot();
});

it('should render Footer', () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});

it('should render Histories', () => {
  const { container } = render(<Histories histories={mockHistories} />);
  expect(container).toMatchSnapshot();
});

it('should render Menu', () => {
  const { container } = render(<Menu username="username" />);
  expect(container).toMatchSnapshot();
});

it('should render Projects', () => {
  const { container } = render(<Projects experiences={mockExperiences} />);
  expect(container).toMatchSnapshot();
});

it('should render Skills', () => {
  const { container } = render(<Skills skills={mockSkills} />);
  expect(container).toMatchSnapshot();
});

it('should render Articles', () => {
  const { container } = render(<Articles />);
  expect(container).toMatchSnapshot();
});
