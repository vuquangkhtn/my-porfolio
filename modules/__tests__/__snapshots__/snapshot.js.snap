// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render About 1`] = `
<div>
  <section
    class="about full-screen d-lg-flex justify-content-center align-items-center"
    id="about"
    test-id="about"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-7 col-md-12 col-12 d-flex align-items-center"
        >
          <div
            class="about-text"
          >
            <small
              class="small-text"
            >
              Welcome to
               
              <span
                class="mobile-block"
              >
                my portfolio website!
              </span>
            </small>
            <h1
              class="animated animated-text"
            >
              <span
                class="mr-2"
              >
                Hey folks, I am
              </span>
              <div
                class="animated-info"
              >
                <span
                  class="animated-item"
                >
                  Quang
                </span>
                <span
                  class="animated-item"
                >
                  Web developer
                </span>
              </div>
            </h1>
            <p>
              user bio text
            </p>
            <div
              class="custom-btn-group mt-4"
            >
              <button
                class="sc-bdvvtL dVMeJj mr-lg-2 mb-2"
                mode="primary"
              >
                <div
                  class="d-flex align-items-center"
                >
                  <span />
                   
                  <span>
                    Download Resume
                  </span>
                </div>
              </button>
              <button
                class="sc-bdvvtL exrfkh ml-2 mb-2"
                mode="secondary"
              >
                Get my contact
              </button>
            </div>
          </div>
        </div>
        <div
          class="col-lg-5 col-md-12 col-12"
        >
          <div
            class="about-image svg"
          />
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`should render Articles 1`] = `
<div>
  <section
    class="resume py-5 d-lg-flex justify-content-center align-items-center"
    id="education"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h2
            class="mb-4"
          >
            Articles
          </h2>
          <div
            class="row"
          >
            <div
              class="col-md-6 col-sm-12 col-lg-4 py-3 pe-auto"
            >
              <div
                class="card mr-1"
              >
                <div
                  class="card-header"
                  style="height: 200px; overflow: hidden;"
                />
                <div
                  class="card-body"
                >
                  <h4
                    class=""
                  >
                    How to manage states in React applications effectively
                  </h4>
                  <p
                    class="card-text"
                  >
                    you can keep not only a clean and readable state management but also reduce the complex logic handling in your applications.
                  </p>
                </div>
              </div>
            </div>
            <div
              class="col-md-6 col-sm-12 col-lg-4 py-3 pe-auto"
            >
              <div
                class="card mr-1"
              >
                <div
                  class="card-header"
                  style="height: 200px; overflow: hidden;"
                />
                <div
                  class="card-body"
                >
                  <h4
                    class=""
                  >
                    My Porfolio Common Library
                  </h4>
                  <p
                    class="card-text"
                  >
                    a React UI library with a comprehensive set of components for My Porfolio page.
                  </p>
                </div>
              </div>
            </div>
            <div
              class="col-md-6 col-sm-12 col-lg-4 py-3 pe-auto"
            >
              <div
                class="card mr-1"
              >
                <div
                  class="card-header"
                  style="height: 200px; overflow: hidden;"
                />
                <div
                  class="card-body"
                >
                  <h4
                    class=""
                  >
                    My Porfolio Server
                  </h4>
                  <p
                    class="card-text"
                  >
                    a server using Graphql which store and dispatch data for My Porfolio page.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`should render Contact 1`] = `
<div>
  <section
    class="contact py-5"
    id="contact"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-5 mr-lg-5 col-12"
        >
          <div
            class="google-map w-100"
          >
            <iframe
              allowfullscreen=""
              frameborder="0"
              height="300"
              src="https://maps.google.com/maps?q=Mock%20Address&t=&z=13&ie=UTF8&iwloc=&output=embed"
              style="border: 0px;"
              width="400"
            />
          </div>
          <div
            class="contact-info d-flex justify-content-between align-items-center py-4 px-lg-5"
          >
            <div
              class="contact-info-item"
            >
              <h3
                class="mb-3 text-white"
              >
                My information
              </h3>
              <p
                class="footer-text mb-0"
              >
                +(84)123 456 789
              </p>
              <p>
                <a
                  href="mailto:vuquang@mockemail.com"
                >
                  vuquang@mockemail.com
                </a>
              </p>
            </div>
            <ul
              class="social-links"
            >
              <li>
                <a
                  href="github.com/vuquang"
                  rel="noreferrer"
                  target="_blank"
                  title="github"
                >
                  <span />
                </a>
              </li>
              <li>
                <a
                  href="linkedin.com/vuquang"
                  rel="noreferrer"
                  target="_blank"
                  title="linkedin"
                >
                  <span />
                </a>
              </li>
              <li>
                <a
                  href="tel:84123456789"
                  title="phone"
                >
                  <span />
                </a>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="col-lg-6 col-12"
        >
          <div
            class="contact-form"
          >
            <h2
              class="mb-4"
            >
              Interested to work together? Let's talk
            </h2>
            <div>
              <div
                class="row"
              >
                <div
                  class="col-lg-6 col-12"
                >
                  <input
                    autocomplete="nope"
                    class="sc-gsDKAQ bOKthw"
                    name="name"
                    placeholder="Your Name"
                    type="text"
                  />
                </div>
                <div
                  class="col-lg-6 col-12"
                >
                  <input
                    autocomplete="nope"
                    class="sc-gsDKAQ bOKthw"
                    name="email"
                    placeholder="Email"
                    type="email"
                  />
                </div>
                <div
                  class="col-12"
                >
                  <textarea
                    autocomplete="nope"
                    class="sc-dkPtRN eftWWY"
                    name="message"
                    placeholder="Message"
                    rows="6"
                  />
                </div>
                <div
                  class="col-lg-5 col-12"
                >
                  <button
                    class="sc-bdvvtL gLFEMN"
                    mode="submit"
                  >
                    Send Contact
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`should render Educations 1`] = `
<div>
  <section
    class="resume py-5 d-lg-flex justify-content-center align-items-center"
    id="education"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h2
            class="mb-4"
          >
            Educations
          </h2>
          <div
            class="timeline"
          >
            <div
              class="timeline-wrapper"
            >
              <div
                class="timeline-yr"
              >
                <span>
                  Organization 1
                </span>
              </div>
              <div
                class="timeline-info"
              >
                <h3>
                  <span />
                  Education 1
                </h3>
                <p>
                  Description 1
                </p>
              </div>
            </div>
            <div
              class="timeline-wrapper"
            >
              <div
                class="timeline-yr"
              >
                <span>
                  Organization 2
                </span>
              </div>
              <div
                class="timeline-info"
              >
                <h3>
                  <span />
                  Education 2
                </h3>
                <p>
                  Description 2
                </p>
              </div>
            </div>
            <div
              class="timeline-wrapper"
            >
              <div
                class="timeline-yr"
              >
                <span>
                  Organization 3
                </span>
              </div>
              <div
                class="timeline-info"
              >
                <h3>
                  <span />
                  Education 3
                </h3>
                <p>
                  Description 3
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`should render Footer 1`] = `
<div>
  <footer
    class="footer py-5"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-lg-12 col-12"
        >
          <p
            class="copyright-text text-center"
          >
            Designed by
             
            <a
              class="sc-hKwDye hjOJSd"
              href="htwtps://www.facebook.com/tooplate"
              rel="nofollow"
            >
              Tooplate
            </a>
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;

exports[`should render Histories 1`] = `
<div>
  <section
    class="resume py-5 d-lg-flex justify-content-center align-items-center"
    id="history"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h2
            class="mb-4"
          >
            Histories
          </h2>
          <div
            class="timeline"
          >
            <div
              class="timeline-wrapper"
            >
              <div
                class="timeline-yr"
              >
                <span>
                  12 mths
                </span>
              </div>
              <div
                class="timeline-info"
              >
                <h3>
                  <span />
                  His Name 1
                  <small>
                    Company 1
                  </small>
                </h3>
                <p>
                  From 10/2020 to 10/2021
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`should render Menu 1`] = `
<div>
  <nav
    class="navbar navbar-expand-sm navbar-light headroom"
  >
    <div
      class="container"
    >
      <a
        class="navbar-brand"
        href=""
      >
        <span />
         
        username
      </a>
      <button
        class="sc-eCImPb ifRcVo navbar-toggler"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
        <span
          class="navbar-toggler-icon"
        />
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
      >
        <ul
          class="navbar-nav mx-auto"
        >
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#skill"
            >
              <span
                data-hover="Skill"
              >
                Skill
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#history"
            >
              <span
                data-hover="History"
              >
                History
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#project"
            >
              <span
                data-hover="Project"
              >
                Project
              </span>
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              class="nav-link"
              href="#contact"
            >
              <span
                data-hover="Contact"
              >
                Contact
              </span>
            </a>
          </li>
        </ul>
        <ul
          class="navbar-nav ml-lg-auto"
        >
          <div
            class="ml-lg-4"
          >
            <div
              class="color-mode d-lg-flex justify-content-center align-items-center"
              data-testid="light-icon"
            >
              <span />
              Color mode
            </div>
          </div>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`should render Projects 1`] = `
<div>
  <section
    class="resume py-5 d-lg-flex justify-content-center align-items-center"
    id="project"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h2
            class="mb-4"
          >
            Project Experiences
          </h2>
          <div
            class="timeline"
          >
            <div
              class="timeline-wrapper"
            >
              <div
                class="timeline-yr"
              >
                <span>
                  1
                </span>
              </div>
              <div
                class="timeline-info"
              >
                <h3>
                  <span>
                    Software Eng
                  </span>
                  <small>
                    Web
                  </small>
                </h3>
                <p>
                  Description for Exp
                </p>
                <b>
                  skills 1
                </b>
              </div>
            </div>
            <div
              class="timeline-wrapper"
            >
              <div
                class="timeline-yr"
              >
                <span>
                  2
                </span>
              </div>
              <div
                class="timeline-info"
              >
                <h3>
                  <span>
                    Web Developer
                  </span>
                  <small>
                    Website
                  </small>
                </h3>
                <p>
                  Description for Exp 2
                </p>
                <b>
                  skill 2
                </b>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`should render Skills 1`] = `
<div>
  <section
    class="resume py-5 d-lg-flex justify-content-center align-items-center"
    id="skill"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <h2
            class="mb-4"
          >
            Skills
          </h2>
          <div
            data-testid="skill-list"
          >
            <div
              class="d-lg-flex align-items-center mb-3"
            >
              <h4
                class="mb-1 mr-2"
              >
                <span>
                  Skill 1
                </span>
              </h4>
              <div
                class="d-flex flex-wrap"
                style="row-gap: 5px;"
              >
                <div
                  class="medal mr-1"
                >
                  Skill Detail 1
                </div>
              </div>
            </div>
            <div
              class="d-lg-flex align-items-center mb-3"
            >
              <h4
                class="mb-1 mr-2"
              >
                <span>
                  Skill 2
                </span>
              </h4>
              <div
                class="d-flex flex-wrap"
                style="row-gap: 5px;"
              >
                <div
                  class="medal mr-1"
                >
                  Skill Detail 2
                </div>
              </div>
            </div>
            <div
              class="d-lg-flex align-items-center mb-3"
            >
              <h4
                class="mb-1 mr-2"
              >
                <span>
                  Skill 3
                </span>
              </h4>
              <div
                class="d-flex flex-wrap"
                style="row-gap: 5px;"
              >
                <div
                  class="medal mr-1"
                >
                  Skill Detail 3
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
